astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -23,212 +147,509 @@ class NDArithmeticMixin(object):
+    interface you can alter the ``propagate_uncertainties`` parameter in
+    :meth:`NDArithmeticMixin.add`. ``None`` means that the result will have no
+    uncertainty, ``False`` means it takes the uncertainty of the first operand
+    (if this does not exist from the second operand) as the results
+    uncertainty. This behaviour is also explained in the help-page for the
+    different arithmetic operations.
+
+    Notes
+    -----
+    1. It is not tried to decompose the units, mainly due to the internal
+       mechanics of `~astropy.units.Quantity`, so the resulting data might have
+       units like ``km/m`` if you divided for example 100km by 5m. So this
+       Mixin has adopted this behaviour.
+
+    Examples
+    --------
+
+    TODO: Just here because the examples fit more into the main documentation
+    than in here.
+
+    For example::
+
+        >>> from astropy.nddata import *
+        >>> class NDDataWithMath(NDArithmeticMixin, NDData): pass
+        >>> nd = NDDataWithMath([1,2,3], unit='meter')
+        >>> nd_inv = nd.ic_division(1, nd)
+        >>> nd_inv.__class__.__name__
+        'NDDataWithMath'
+        >>> nd_inv.data
+        array([ 1.        ,  0.5       ,  0.33333333])
+        >>> nd_inv.unit
+        Unit("1 / m")
+
+    This method also allows that the result of unrelated objects is tried
+    with what this class implements as arithmetics and give a result that
+    is the same class as the class that called the method.
+
+        >>> nd2 = nd.ic_multiplication([1,2],3)
+        >>> nd2.__class__.__name__
+        'NDDataWithMath'
+        >>> nd2.data
+        array([3, 6])
+
+    Since this is a classmethod we don't need an instance to use them:
+
+        >>> nd3 = NDDataWithMath.ic_subtraction(5, NDData([1,2,3]))
+        >>> nd3.__class__.__name__
+        'NDDataWithMath'
+        >>> nd3.data
+        array([4, 3, 2])
+
+    And it allows to handle arithmetics with different subclasses.
+
+        >>> class NDDataWithMathAndSlicing(NDSlicingMixin,NDArithmeticMixin, NDData): pass
+        >>> nd = NDDataWithMath([5,5,5])
+        >>> nd2 = NDDataWithMathAndSlicing([3,2,5])
+        >>> nd3 = nd2.ic_addition(nd, nd2)
+        >>> nd3.__class__.__name__
+        'NDDataWithMathAndSlicing'
+        >>> nd3.data
+        array([ 8,  7, 10])
     """
-    def _arithmetic(self, operand, propagate_uncertainties, name, operation):
+
+    def _arithmetic(self, operation, operand,
+                    propagate_uncertainties=True, handle_mask=np.logical_or,
+                    handle_meta=False, uncertainty_correlation=0,
+                    compare_wcs=False, **kwds):
         """
-        {name} another dataset (``operand``) to this dataset.
+        Base method which calculates the result of the arithmetic operation.
+
+        This class determines the result of the arithmetic operation on the
+        ``data`` with respect to their units and then forwards to other methods
```

Comment:
Do you mean "this method" instead of "this class"? Also, I think "with respect to" → "including"